src/main/java/com/iexec/core/result/ResultService.java

```diff
@@ -33,27 +35,28 @@
 @Service
 public class ResultService {
 
-    private final ResultProxyClient resultProxyClient;
+    private final ResultRepositoryConfiguration resultRepositoryConfiguration;
     private final SignatureService signatureService;
     private final TaskService taskService;
 
-    public ResultService(final ResultProxyClient resultProxyClient, final SignatureService signatureService, final TaskService taskService) {
-        this.resultProxyClient = resultProxyClient;
+    public ResultService(final ResultRepositoryConfiguration resultRepositoryConfiguration, final SignatureService signatureService, final TaskService taskService) {
+        this.resultRepositoryConfiguration = resultRepositoryConfiguration;
         this.signatureService = signatureService;
         this.taskService = taskService;
     }
 
     @Retryable(value = FeignException.class)
-    public boolean isResultUploaded(final String chainTaskId) {
-        final String enclaveChallenge = taskService.getTaskByChainTaskId(chainTaskId).map(Task::getEnclaveChallenge).orElse(EMPTY_ADDRESS);
-        final WorkerpoolAuthorization workerpoolAuthorization = signatureService.createAuthorization(signatureService.getAddress(), chainTaskId, enclaveChallenge);
+    public boolean isResultUploaded(final TaskDescription task) {
```

Comment:
You don't know of it yet, but a `Task` class exists in scheduler.
This task describes tasks as stored in MongoDB.
Thus, to avoid ambiguous reasoning, we will prefer:
```suggestion
    public boolean isResultUploaded(final TaskDescription taskDescription) {
```